### PR TITLE
vdk-core: step loading failure misclassified as Platform error fix

### PR DESCRIPTION
**Pull request by ivakoleva**
Cases of any data job step source cannot be loaded, due to syntax error,
or issue with importing dependency, were misclassified as
Platform error.

Failure to load step source, is now classified correctly as User error.
 
Testing Done: added a test to cover the use-case.

### CODE REVIEW

File: projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py [hunk 1]
```diff
@@ -74,8 +74,27 @@ def run(job_input)
...
             filename = step.file_path.name
             namespace = "step_" + str(filename).lower().strip(".py")
-            python_module = imp.load_source(namespace, str(step.file_path))
             success = False
+
+            try:
+                log.debug("Loading %s ..." % filename)
+                python_module = imp.load_source(namespace, str(step.file_path))
+                log.debug("Loading %s SUCCESS" % filename)
+            except BaseException as e:
```
Comment (doks5) [inline projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:83]: Is there a specific reason for catching BaseException and not just Exception?

File: projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py [hunk 2]
```diff
@@ -74,8 +74,27 @@ def run(job_input)
             sys.path.insert(0, str(step.job_dir))
             filename = step.file_path.name
             namespace = "step_" + str(filename).lower().strip(".py")
-            python_module = imp.load_source(namespace, str(step.file_path))
             success = False
+
+            try:
+                log.debug("Loading %s ..." % filename)
```
Comment (doks5) [inline projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:80]: nit: Why not using f strings?

**Review by doks5 — APPROVED** [review]

File: projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py (hunk 2 above)
Comment (ivakoleva) [reply projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:80]: for consistency with the same usage, several lines bellow:
`log.info("Entering %s#run(...) ..." % filename)`

File: projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py (hunk 1 above)
Comment (ivakoleva) [reply projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:83]: saw same approach in `cli_run.py` and `data_job.py` on the way the exception propagates

File: projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py (hunk 2 above)
Comment (doks5) [reply projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:80]: Gotcha, ok.

File: projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py (hunk 1 above)
Comment (doks5) [reply projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:83]: Maybe we need to consider how we construct exceptions. According to the documentation, BaseException is not meant to be inherited from https://docs.python.org/3/library/exceptions.html#BaseException, so in theory, we shouldn't expect to need to catch any exception at that low level

File: projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py (hunk 2 above)
Comment (tozka) [reply projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:80]: Note that some of those lines were written 4 years ago or more before "f-strings". So we should switch them all over to f-strings when we see opportunity.

**Pull request merged**